COMPATIBILITY.md

```diff
@@ -0,0 +1,17 @@
+# Compatibility
+
+Volta currently tests against the following platforms, and will treat it as a breaking change to drop support for them:
+
+- macOS
+    - x86-64
+    - Apple Silicon
+- Linux x86-64
+- Windows x86-64
+
+We compile release artifacts compatible with the following, and likewise will treat it as a breaking change to drop support for them:
```

Comment:
While not written down, this has _effectively_ been our policy up until now. It's definitely caused some complications, especially around maintaining the CI / release GitHub actions.

I'm generally in favor of changing the supported versions being a breaking change. I think we should be aware of how often we find ourselves needing to bump the major, to avoid churn, but up till now that hasn't been an issue.

I also wonder if it's worth explicitly adopting a model similar to Ember's, where our major changes are mostly added in backwards-compatible minor versions, and the Major bumps are explicitly only for removing deprecated features, dropping support for old OSes, etc.

Reply:
Both of those instincts—keep an eye on making sure breaking releases are rare, and only use them for actually breaking changes—match my own. That being said, we should think hard about our “messaging” strategy around that; “Volta 2.0 has no new features in it!” definitely *will* confuse people, from our experience with Ember. I still like it and think it is good, but because versioning does double duty when using SemVer (communication about changes *and* marketing), and because the marketing version number is an inescapable reality because that’s how the entire rest of the ecosystem uses it (and has for decades!), we will want a message about our thinking there.